src/desdemona/plugins/swift.clj

```diff
@@ -0,0 +1,127 @@
+(ns desdemona.plugins.swift
+  (:require
+   [onyx.peer.pipeline-extensions :as p-ext]
+   [onyx.peer.function :as function]
+   [taoensso.timbre :refer [info]]
+   [cheshire.core :as json]
+   [clj-http.client :as http]
+   [clj-time.core :as t]
+   [clj-time.format :as f]
+   [byte-streams :as bs]
+   [camel-snake-kebab.core :refer [->kebab-case-keyword]]))
+
+(defn ^:private find-first
+  "Get the first item from the given collection for which the given function
+  returns true."
+  [f coll]
+  (first (filter f coll)))
+
+(defn get-cloud-files
+  "Reach into an authentication response and extract the Cloud Files data from
+  the service catalog."
+  [auth-response]
+  (find-first #(and (= "cloudFiles" (:name %)) (= "object-store" (:type %)))
+              (-> auth-response :access :service-catalog)))
+
+(defn get-token
+  "Reach into an authentication response and extract the auth-related part
+  of it (which will include both the auth token and the expiration date)."
+  [auth-response]
+  (-> auth-response :access :token))
+
+(defn authenticate
+  "Authenticate with Cloud Files and return the authentication response. This
+  response will include an auth token that can be used for subsequent
+  requests, as well as a service catalog that will indicate the URL to use
+  for, for example, Cloud Files."
+  [auth-url username api-key]
+  (let [body (json/generate-string {"auth" {"RAX-KSKEY:apiKeyCredentials"
+                                            {"username" username
+                                             "apiKey" api-key}}})
+        response (http/post auth-url
+                            {:body body
+                             :content-type :json
+                             :accept :json
+                             :throw-entire-message true
+                             :as :stream})
+        decoded (json/decode-stream (bs/to-reader (response :body))
```

Comment:
Since you're just going to return this verbatim; consider making it the "body" of the let. Also, that expr would probably look a little nicer with `->`, but that's up to you